source/simulator/core.cc

```diff
@@ -99,6 +102,27 @@ namespace aspect
       signals.edit_finite_element_variables(variables);
       return variables;
     }
+
+    /**
+     * Helper function to construct mapping for the model.
+     * The mapping is given by a degree four MappingQ for the case
+     * of a curved mesh, and a cartesian mapping for a rectangular mesh that is
+     * not deformed. Use a MappingQ1 if the mesh is deformed.
+     * If a free surface is enabled, each mapping is later swapped out for a
+     * MappingQ1Eulerian, which allows for mesh deformation during the
+     * computation.
+     */
+    template <int dim>
+    Mapping<dim> *construct_mapping(const GeometryModel::Interface<dim> &geometry_model,
```

Comment:
The result of this function is used to initialize a `std_cxx11::unique_ptr<Mapping<dim>*>`. Can you just return that? It makes it obvious who owns the object (namely the *caller*).